bundle/jsky.app.ot/src/main/scala/edu/gemini/catalog/ui/QueryResultsFrame.scala

```diff
@@ -0,0 +1,206 @@
+        val hasPM = targets.exists(_.properMotion.isDefined)
+        val pmCols = if (hasPM) pmColumns else Nil
+        val magCols = magColumns.filter(m => bandsInTargets.contains(m.band))
+
+        baseColumnNames ::: pmCols ::: magCols
+      }
+
+      override def getRowCount = targets.length
+
+      override def getColumnCount = columns.size
+
+      override def getColumnName(column: Int): String =
+        columns(column).title
+
+      override def getValueAt(rowIndex: Int, columnIndex: Int) = (targets.isDefinedAt(rowIndex) option {
+        val target = targets(rowIndex)
+        columns.zipWithIndex.find(_._2 == columnIndex).map { c =>
+          c._1.render(target)
+        }
+      }).flatten.orNull
+
+    }
+
+    case class QueryResultsFrame(table: Table) extends Frame with PreferredSizeFrame {
+      title = "Query Results"
+      contents = new BorderPanel() {
+        add(new ScrollPane() {
+          contents = table
+        }, BorderPanel.Position.Center)
+
+      }
+      adjustSize()
+      SizePreference.getPosition(this.getClass).foreach { p =>
+        location = p
+      }
+
+      listenTo(this)
+      reactions += {
+        case _: UIElementResized =>
+          SizePreference.setDimension(getClass, Some(this.size))
+        case _: UIElementMoved =>
+          SizePreference.setPosition(getClass, Some(this.location))
+      }
+    }
+
+    val resultsTable = new Table() with SortableTable {
+      private val m = TargetsModel(Nil)
+      model = model
+      val sorter = new TableRowSorter[TargetsModel](m)
+      peer.setRowSorter(sorter)
+      peer.getRowSorter.toggleSortOrder(0)
+
+      // Align Right
+      peer.setDefaultRenderer(classOf[String], new DefaultTableCellRenderer() {
+        setHorizontalAlignment(SwingConstants.RIGHT)
+      })
+
+    }
+    val frame = QueryResultsFrame(resultsTable)
+  }
+
+  private def reloadSearchData(query: CatalogQuery) {
+    import QueryResultsWindow.table._
+
+    VoTableClient.catalog(query).onComplete {
+      case _: scala.util.Failure[_] =>
+        GlassLabel.hide(frame.peer.getRootPane) // TODO Display error
+      case scala.util.Success(x) if x.result.containsError =>
+        GlassLabel.hide(frame.peer.getRootPane) // TODO Display error
+      case scala.util.Success(x) =>
+        Swing.onEDT {
+          GlassLabel.hide(frame.peer.getRootPane)
+          val model = TargetsModel(x.result.targets.rows)
+          resultsTable.model = model
+
+          val sorter = new TableRowSorter[TargetsModel](model)
+          resultsTable.peer.setRowSorter(sorter)
+          resultsTable.peer.getRowSorter.toggleSortOrder(0)
+        }
+    }
+  }
+
+  // Public interface
+  val instance = this
+
+  private def showTable(q: CatalogQuery):Unit = Swing.onEDT {
+    import QueryResultsWindow.table._
+    if (frame.visible) {
+      frame.peer.toFront()
+    } else {
+      frame.visible = true
+      frame.peer.toFront()
+    }
```

Comment:
Is this much different than eliminating the `if` statement and just always doing the `else` branch?